client/web/src/repo/tree/TreePagePanels.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useCallback, useRef, useState } from 'react'
+
+import { mdiFileDocumentOutline, mdiFolderOutline, mdiMenuUp, mdiMenuDown } from '@mdi/js'
+import classNames from 'classnames'
+import * as H from 'history'
+
+import { LinkOrSpan } from '@sourcegraph/shared/src/components/LinkOrSpan'
+import { TreeFields } from '@sourcegraph/shared/src/graphql-operations'
+import { StackedMeter, Link, Icon, Card, CardHeader, ParentSize } from '@sourcegraph/wildcard'
+
+import { RenderedFile } from '../blob/RenderedFile'
+
+import styles from './TreePage.module.scss'
+
+interface ReadmePreviewCardProps {
+    readmeHTML: string
+    readmeURL: string
+    location: H.Location
+}
+
+export const ReadmePreviewCard: React.FunctionComponent<ReadmePreviewCardProps> = ({
+    readmeHTML,
+    readmeURL,
+    location,
+}) => {
+    const fileRef = useRef<HTMLDivElement>(null)
+    const containerRef = useRef<HTMLDivElement>(null)
+    const isNotCutoff =
+        fileRef.current &&
+        containerRef.current &&
+        fileRef.current.clientHeight > 0 &&
+        containerRef.current.clientHeight >= fileRef.current.clientHeight
+    return (
+        <>
+            <div className={classNames(styles.readmeContainer)} ref={containerRef}>
+                <div ref={fileRef}>
+                    <RenderedFile className={styles.readme} dangerousInnerHTML={readmeHTML} location={location} />
+                </div>
+                <div
+                    className={
+                        isNotCutoff
+                            ? classNames(styles.readmeFader, styles.readmeFaderInvisible)
+                            : classNames(styles.readmeFader)
+                    }
+                />
+            </div>
+            <div
+                className={
+                    isNotCutoff
+                        ? classNames(styles.readmeMore)
+                        : classNames(styles.readmeMore, styles.readmeMoreVisible)
+                }
+            >
+                <Link to={readmeURL}>More...</Link>
```

Comment:
It opens a new page view so IMO:

```suggestion
                <Link to={readmeURL}>View full file</Link>
```